wcfsetup/install/files/lib/util/PasswordUtil.class.php

```diff
@@ -332,8 +332,116 @@ protected static function mybb1($username, $password, $salt, $dbHash) {
 	 * @param	string		$dbHash
 	 * @return	boolean
 	 */
-	protected static function phpbb3($username, $password, $salt, $dbHash) {
-		return self::phpass($username, $password, $salt, $dbHash);
+	public static function phpbb3($username, $password, $salt, $dbHash) {
+		$phpassResult = self::phpass($username, $password, $salt, $dbHash);
+		
+		if ($phpassResult === true) {
+			return true;
+		}
+		
+		if (!preg_match('/^\$([^$]+)\$/', $dbHash, $matches)) {
+			return false;
+		}
+		
+		$algorithms = explode('\\', $matches[1]);
+		// Strip the type prefix.
+		$dbHash = substr($dbHash, strlen($matches[0]));
+		
+		// The following loop only supports the multi-hash variant.
+		// Everything else should already be handled at this point.
+		if (count($algorithms) == 1) {
+			return false;
+		}
+		foreach ($algorithms as $algorithm) {
+			if (($dollar = strpos($dbHash, '$')) === false) {
```

Comment:
```suggestion
			$dollar = strpos($dbHash, '$')
			if ($dollar === false) {
```
The value is required later on, making it easier to find it instead of hiding it in an `if`.